inc/saihostintf.h

```diff
@@ -353,35 +372,14 @@ typedef enum _sai_hostif_trap_attr_t
     SAI_HOSTIF_TRAP_ATTR_TRAP_PRIORITY,
 
     /**
-     * @brief Trap channel to use
-     *
-     * @type sai_hostif_trap_channel_t
-     * @flags CREATE_AND_SET
-     * @default SAI_HOSTIF_TRAP_CHANNEL_CB
-     */
-    SAI_HOSTIF_TRAP_ATTR_TRAP_CHANNEL,
-
-    /**
-     * @brief file descriptor
-     *
-     * Valid only when #SAI_HOSTIF_TRAP_ATTR_TRAP_CHANNEL == #SAI_HOSTIF_TRAP_CHANNEL_FD
-     * Must be set before set #SAI_HOSTIF_TRAP_ATTR_TRAP_CHANNEL to #SAI_HOSTIF_TRAP_CHANNEL_FD
-     *
-     * @type sai_object_id_t
-     * @flags CREATE_AND_SET
-     * @objects SAI_OBJECT_TYPE_HOSTIF
-     */
-    SAI_HOSTIF_TRAP_ATTR_FD,
-
-    /**
-     * @brief Enable trap for a list of SAI ports
+     * @brief List of SAI ports to be excluded (disabled) from the trap generation
      *
      * @type sai_object_list_t
      * @objects SAI_OBJECT_TYPE_PORT
      * @flags CREATE_AND_SET
      * @default empty
      */
-    SAI_HOSTIF_TRAP_ATTR_PORT_LIST,
+    SAI_HOSTIF_TRAP_ATTR_EXCLUDE_PORT_LIST,
```

Comment:
The motivation is that the most common use case is to have the trap enabled for all the ports
With the old attribute, port list, the default is empty list. So the application had to create a list of all the ports and provide it for this use case. In addition, whenever breaking out a port or creating a LAG, the application would have had to update this list.
With the new attribute, exclude port list, it is very easy to address the global use case. Just provide an empty list of ports to exclude, and the trap will be enabled for all ports. 
And still, with the new attribute, it is possible to have the trap enabled only for a subset of ports.